packages/launcher/lib/linux/index.ts

```diff
@@ -36,22 +44,30 @@ function getLinuxBrowser (
     throw notInstalledErr(binary)
   }
 
+  const maybeSetSnapProfilePath = (versionString: string) => {
+    if (os.platform() === 'linux' && name === 'chromium' && versionString.endsWith('snap')) {
+      // when running as a snap, chromium can only write to certain directories
+      // @see https://github.com/cypress-io/cypress/issues/7020
+      foundBrowser.profilePath = path.join(os.homedir(), 'snap', 'chromium', 'current')
```

Comment:
It'd be great to also add a debug log here to know we're modifying the path because of this.